src/XCCDF/xccdf_session.c

```diff
@@ -455,18 +471,30 @@ bool xccdf_session_set_profile_id_by_suffix(struct xccdf_session *session, const
 			const char *bench_profile_id = xccdf_profile_get_id(bench_profile);
 
 			if(oscap_str_endswith(bench_profile_id, profile_suffix)) {
-				full_profile_id = bench_profile_id;
-				break;
+				if (full_profile_id != NULL) {
+					printf("Matches: \n%s\n", full_profile_id);
+					printf("%s\n", bench_profile_id);
+					multiple = true;
+					break;
+				}
+				else {
+					full_profile_id = bench_profile_id;
+				}
 			}
 		}
-
 		xccdf_profile_iterator_free(profit_bench);
 	}
 
-	if (full_profile_id == NULL)
-		return false;
-	else
-		return xccdf_session_set_profile_id(session, full_profile_id);
+	if (full_profile_id == NULL) {
+		return 1;
+	}
+	else if (multiple) {
+		return 2;
+	}
+	else {
+		xccdf_session_set_profile_id(session, full_profile_id);
```

Comment:
on second thought I think we should assert that it returns 0 and then return 0. because if the code is correct this must always return 0.

Please change the line to:
`assert(xccdf_session_set_profile_id(session, full_profile_id) == 0);`